### DISCUSSION TITLE
MenuItem:pointerover setting foreground color

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#17037] [Q&A] [answered]
(Avalonia 11.1.2 on Windows) Currently I have a custom MenuItem class ('MainMenuHeader' - background very dark blue) for the top-level menu items, and I can set the foreground color fine for the 'normal' state, but I can't change the foreground color in the pointerover state (the default black foreground in the pointerover state doesn't show on a dark menu bar):

```
		<Style Selector="MenuItem.MainMenuHeader">
			<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderForeColor}"/>
		</Style>

		<Style Selector="MenuItem.MainMenuHeader:pointerover /template/ ContentPresenter#PART_ContentPresenter">
			<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderHoverColor}"/>
		</Style>

```
The 2 colors are defined as:

```
			<SolidColorBrush x:Key='MainMenuHeaderForeColor'>#FBFFFF</SolidColorBrush>
			<SolidColorBrush x:Key='MainMenuHeaderHoverColor'>#B6DADA</SolidColorBrush>

```

**Comment by TangDave** [comment]
Also tried the following, which also had no result:

		<Style Selector="MenuItem.MainMenuHeader">
			<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderForeColor}"/>

			<Style Selector="^:pointerover">
				<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderHoverForeColor}"/>
			</Style>
		</Style>

**Comment by TangDave** [comment]
Ditto for:

	<Style Selector="MenuItem.MainMenuHeader">
		<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderForeColor}"/>

		<Style Selector="^:pointerover /template/ ContentPresenter#PART_ContentPresenter">
			<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderHoverForeColor}"/>
		</Style>
	</Style>

**Comment by TangDave** [comment, marked as answer]
Found the answer in the MenuItem.xaml file:

```
<Style Selector="MenuItem.MainMenuHeader">
	<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderForeColor}"/>

	<Style Selector="^:pointerover /template/ ContentPresenter#PART_HeaderPresenter">
		<Setter Property="Foreground" Value="{DynamicResource MainMenuHeaderHoverForeColor}"/>
	</Style>
</Style>
```

**Reply by Timskt** [reply]
thank you
